src/pydocstyle/violations.py

```diff
@@ -68,9 +79,10 @@ def lines(self):
                 break
         return source
 
-    def __str__(self):
-        self.explanation = '\n'.join(l for l in self.explanation.split('\n')
-                                     if not is_blank(l))
+    def __str__(self) -> str:
+        if self.explanation:
+            self.explanation = '\n'.join(l for l in self.explanation.split('\n')
```

Comment:
Unrelated to this PR, but shouldn't this only happen once?

Reply:
In a different PR :)